service_capacity_modeling/capacity_planner.py

```diff
@@ -624,6 +647,12 @@ def generate_scenarios(  # pylint: disable=too-many-positional-arguments
         # Set current instance object if exists
         _set_instance_objects(desires, hardware)
 
+        # We should not even bother with shapes that don't meet the minimums
+        (
+            per_instance_cores,
+            per_instance_mem,
+        ) = self._per_instance_requirements(desires)
```

Comment:
nit: Does it make sense to materialize this tuple in a model? The composite feels domain relevant.

Reply:
hmm perhaps? This was there before, it was just specific to stateful desires (i.e. read_threads -> cpus). I generalized it here. Since it was in use before, I imagine that in practice most models agreed on this conversion.

Additionally, models have a way to bypass this completely.